src/main/java/com/google/sps/servlets/ChallengesServlet.java

```diff
@@ -13,38 +13,130 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.gson.*;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.json.JsonFactory;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultList;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
 
-import com.google.gson.Gson;
 import com.google.sps.data.Challenge;
 import com.google.sps.data.ChallengeData;
+import com.google.sps.data.User;
+
 import java.io.IOException;
 import java.net.URL;
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Scanner;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import javafx.util.Pair;
+
 /** Servlet that manages challenges **/
 @WebServlet("/challenges")
 public class ChallengesServlet extends HttpServlet {
+  private static final String CLIENT_ID = "605480199600-e4uo1livbvl58cup3qtd1miqas7vspcu.apps.googleusercontent.com";
```

Comment:
Made a GoogleIdHelper class that you can use for verification so feel free to use that if you're interested -- ideally will help with testing the servlets that require verification
If you want to use it, you can remove lines 62-67 (the client id through verifier) and remove the verifyId function and replace lines 77-84 with:
```
Payload payload = GoogleIdHelper.verifyId(request);
   if (payload == null) {
        response.setStatus(400);
        return;
    }
```